cmd/snap-confine/udev-support.c

```diff
@@ -18,317 +18,436 @@
 
 #include <ctype.h>
 #include <errno.h>
-#include <limits.h>
-#include <sys/sysmacros.h>
-#include <sched.h>
+#include <fcntl.h>
 #include <string.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
-#include <sys/wait.h>
 #include <unistd.h>
 
+#include <libudev.h>
+
+#include "../libsnap-confine-private/cleanup-funcs.h"
 #include "../libsnap-confine-private/snap.h"
 #include "../libsnap-confine-private/string-utils.h"
+#include "../libsnap-confine-private/cgroup-support.h"
 #include "../libsnap-confine-private/utils.h"
 #include "udev-support.h"
 
-static void
-_run_snappy_app_dev_add_majmin(struct snappy_udev *udev_s,
-			       const char *path, unsigned major, unsigned minor)
+__attribute__((format(printf, 2, 3)))
+static void sc_dprintf(int fd, const char *format, ...);
+
+static void sc_dprintf(int fd, const char *format, ...)
```

Comment:
/me idly notes that perhaps the udev-support file is not the best place for this generic utility function given that we already have a `utils.h`, but that's a nitpick, better to land this PR before it's first birthday and handle in followups

Reply:
This is here for a reason but that has been lot to time. This is the current version, in v2 support mode this changes to a more complex machinery that is still local to this file. I would keep it here for now, as it is a step along the path.